pkg/api/validation/validation.go

```diff
@@ -536,6 +536,10 @@ func validatePullPolicy(ctr *api.Container) errs.ValidationErrorList {
 func validateContainers(containers []api.Container, volumes util.StringSet) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 
+	if len(containers) == 0 {
+		return append(allErrs, errs.NewFieldRequired(""))
```

Comment:
This is not needed, because all errors returned from this method will be prefixed with "containers" so the error will be like:

[spec.template.spec.containers: required value]

(prefix may vary depending on where the PodSpec is used)